neo/Implementations/Blockchains/LevelDB/LevelDBBlockchain.cs

```diff
@@ -685,6 +685,7 @@ private void PersistBlocks()
                         // Reset cancellation token.
                         VerificationCancellationToken = new CancellationTokenSource();
                     }
+                    OnPersistUnlocked(block);
```

Comment:
Maybe is better put this line below the next `lock` (L693), and prevent the event to have wrong `block_cache` access

Reply:
I agree with that. I'll moved it below L693 and updated this PR.